src/Objs/defaultPageLayoutComponent.tsx

```diff
@@ -1,19 +1,96 @@
-import { ContentTag, CurrentPage, provideLayoutComponent } from 'scrivito'
+import {
+  connect,
+  ContentTag,
+  CurrentPage,
+  Obj,
+  provideLayoutComponent,
+} from 'scrivito'
 
 type ObjClass = Parameters<typeof provideLayoutComponent>[0]
 
 export function provideDefaultPageLayoutComponent(objClass: ObjClass) {
   provideLayoutComponent(objClass, ({ page }) => {
+    const showLeftSidebar = !!page.get('layoutShowLeftSidebar')
+    const showRightSidebar = !!page.get('layoutShowRightSidebar')
+    const showSidebar = showLeftSidebar || showRightSidebar
+
     return (
       <>
         {!!page.get('layoutShowHeader') && (
           <ContentTag tag="header" content={page} attribute="layoutHeader" />
         )}
-        <CurrentPage />
+
+        <BackgroundWrapper
+          backgroundColor={page.get('layoutMainBackgroundColor')}
+        >
+          {showSidebar ? (
+            <SidebarLayout
+              page={page}
+              showLeftSidebar={showLeftSidebar}
+              showRightSidebar={showRightSidebar}
+            />
+          ) : (
+            <CurrentPage />
+          )}
+        </BackgroundWrapper>
+
         {!!page.get('layoutShowFooter') && (
           <ContentTag tag="footer" content={page} attribute="layoutFooter" />
         )}
       </>
     )
   })
 }
+
+const BackgroundWrapper = connect(function BackgroundWrapper({
+  backgroundColor,
+  children,
+}: {
+  children: React.ReactNode
+  backgroundColor?: unknown
+}) {
+  if (typeof backgroundColor !== 'string') return children
+  if (backgroundColor === 'transparent') return children
+
+  return <div className={`bg-${backgroundColor}`}>{children}</div>
+})
+
+const SidebarLayout = connect(function SidebarLayout({
+  page,
+  showLeftSidebar,
+  showRightSidebar,
+}: {
+  page: Obj
+  showLeftSidebar: boolean
+  showRightSidebar: boolean
+}) {
+  return (
+    <div className="container py-2">
+      <div className="row">
+        {showLeftSidebar && (
```

Comment:
* [ ] FYI If I switch this on in preview mode it looks wrong. Can we improve the behavior here?

Reply:
I don't know or see what you mean. What does "look wrong" if you switch the left sidebar on?

Reply:
> What does "look wrong" if you switch the left sidebar on?

The space appears on the right-hand side.

<img width="1374" alt="image" src="https://github.com/user-attachments/assets/4befdfd1-8309-40a1-abdc-989dc7f7e9be">
<img width="1374" alt="image" src="https://github.com/user-attachments/assets/aff86654-086b-457e-a6bb-be17eb0f4f52">

Reply:
Ah, sorry I misread "preview" as "edit" mode. I now fixed it.